contracts/core/Controller.sol

```diff
@@ -899,7 +901,14 @@ contract Controller is Initializable, OwnableUpgradeSafe, ReentrancyGuardUpgrade
 
         pool.transferToUser(collateral, _args.to, payout);
 
-        emit VaultSettled(_args.owner, _args.to, address(otoken), _args.vaultId, payout);
+        uint256 vaultId;
+        address payoutRecipient;
+        {
+            vaultId = _args.vaultId;
+            payoutRecipient = _args.to;
+        }
```

Comment:
I think there is no need for this nested scope, it should work fine by declaring the variable directly outside.